Implemented Go Training Tasks

AhmadAwab98 opened the pull request:
### Includes:

- simple CSV to JSON from command line
- using gorilla/mux to create micro service
- using go-chi/chi to create micro service
- web based CSV to JSON
- a JWT authentication in CSV parser
- simple CRUD application using GORM
- casbin authorization in CRUD application

AhmadAwab98 committed:
Convert CSV to JSON

Implemented 
- CSV path parsing from command line
- Conversion of CSV data to JSON format

AhmadAwab98 committed:
Created a simple micro service

- Created handler function
- Added a route
- Tested in the browser

AhmadAwab98 committed:
Created a simple micro service

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Convert CSV to JSON in web

Implemented 
- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Used JWT Authentication to limit access

- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route
- Added authentication for CSV handler
- Added a route for non authenticated users
- Tested in the Postman

AhmadAwab98 committed:
Created a simple CRUD application using gorm

- Created a database and related struct
- Created get all users route and handler
- Created get user by id route and handler
- Created create owner route and handler
- Created update owner route and handler
- Created delete owner route and handler
- Tested in postman
- Exported postman collection in JSON

AhmadAwab98 committed:
Added casbin authorization

- Added Casbin authorization to the crud application
- Made policy.csv and model.conf for casbin authorization
- Tested in postman

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Update README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Created .gitignore

- Created .gitignore to ignore build files

AhmadAwab98 committed:
Created a simple micro service using gorilla/mux

- Created handler function
- Added a route
- Tested in the browser

AhmadAwab98 committed:
Created a simple micro service using gorilla/mux

- Created handler function
- Added a route
- Tested in the browser

AhmadAwab98 committed:
Created a simple micro service using go-chi

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Created a simple micro service using go-chi

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Created a simple micro service using go-chi

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Rename directory

- Renamed directory to CSVtoJSON

AhmadAwab98 committed:
Convert CSV to JSON in web

Implemented 
- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Renamed

- Renamed to GormCRUD

AhmadAwab98 committed:
Created a simple CRUD application using gorm

- Created a database and related struct
- Created get all users route and handler
- Created get user by id route and handler
- Created create owner route and handler
- Created update owner route and handler
- Created delete owner route and handler
- Tested in postman
- Exported postman collection in JSON

AhmadAwab98 committed:
Renamed

- Renamed to casbin

AhmadAwab98 committed:
Added casbin authorization

- Added Casbin authorization to the crud application
- Made policy.csv and model.conf for casbin authorization
- Tested in postman

AhmadAwab98 committed:
Renamed

- Renamed to jwt

AhmadAwab98 committed:
Used JWT Authentication to limit access

- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route
- Added authentication for CSV handler
- Added a route for non authenticated users
- Tested in the Postman

AhmadAwab98 committed:
Renamed

- Renamed to parseCSV

AhmadAwab98 committed:
Convert CSV to JSON

Implemented 
- CSV path parsing from command line
- Conversion of CSV data to JSON format

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Created a simple CRUD application using gorm

- Created a database and related struct
- Created get all users route and handler
- Created get user by id route and handler
- Created create owner route and handler
- Created update owner route and handler
- Created delete owner route and handler
- Tested in postman
- Exported postman collection in JSON

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Convert CSV to JSON in web

Implemented 
- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Renamed

- Renamed to gormcrud

AhmadAwab98 committed:
Created a simple CRUD application using gorm

- Created a database and related struct
- Created get all users route and handler
- Created get user by id route and handler
- Created create owner route and handler
- Created update owner route and handler
- Created delete owner route and handler
- Tested in postman
- Exported postman collection in JSON

AhmadAwab98 committed:
Renamed

- Renamed to web
- Put in a directory

AhmadAwab98 committed:
Renamed

- Renamed to parser
- Moved into a directory

AhmadAwab98 committed:
Convert CSV to JSON

Implemented 
- CSV path parsing from command line
- Conversion of CSV data to JSON format

AhmadAwab98 committed:
Convert CSV to JSON in web

Implemented 
- Created handler function with JSON body of request and response
- CSV path parsing from the request body
- Conversion of CSV data to JSON format
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Created a simple micro service using go-chi

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Created a simple micro service using go-chi

- Created handler function with JSON body of request and response
- Added a route using go-chi
- Tested in the Postman

AhmadAwab98 committed:
Implemented list handler

- Implemented list handler
- docker-compose.yml not created yet

AhmadAwab98 committed:
Merge pull request #9 from AhmadAwab98/feature

Implemented list handler

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Implemented folder and file print

- Implemented listHandler to print folders and files
- Used redis for caching response

AhmadAwab98 committed:
Merge pull request #10 from AhmadAwab98/feature

Implemented list folders

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

fahadsiddiqui commented on folderprint/README.md:
````diff
@@ -0,0 +1,45 @@
+
+- Run the go mod tidy command to add any missing dependencies to the go.mod file
+```
+go mod tidy
````
You don't need to run `go mod init ...`. And `go.mod|sum` files should be part of the project.

fahadsiddiqui commented on folderprint/README.md:
````diff
@@ -0,0 +1,45 @@
+
+- Run the go file
+```
+ go run folderandfileprint.go
````
```
go run ffprint.go
```

fahadsiddiqui commented on folderprint/handler/handler.go:
```diff
@@ -0,0 +1,91 @@
+	// get the response from cache and convert it into json format
+	responsecached, _ := rdb.HGetAll(ctx,"responseCache").Result()
+	err = json.Unmarshal([]byte(responsecached["cachedData"]), &response)
+	JSONresponse, errjson = json.MarshalIndent(response, "", "	")
```
The logic should be as following

```
if the key exists is redis:
    fetch from redis
else
    fetch from real folders
    update redis against MD5 key
```

AhmadAwab98 committed:
Updated .gitignore
	updated:   .gitignore
	renamed:    folderandfileprint.go -> ffprint.go
	modified:   handler/handler.go

AhmadAwab98 committed:
Changes to be committed:
	file added:   go.mod, go.sum

AhmadAwab98 committed:
Updated handler

- Added docker-compose.yml
- Updated handler for the object under folder

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Updated README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Update README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Dockerized whole project

- Added Dockerfile
- Updated docker-compose.yml

AhmadAwab98 committed:
Update README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Update README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file

AhmadAwab98 committed:
Update README.md

Updated README.md to add following details:
- A brief overview of the file
- Execution information of the file
